DSCResources/MSFT_xIPAddress/MSFT_xIPAddress.psm1

```diff
@@ -134,48 +130,162 @@ function ValidateProperties
 
     if(!([System.Net.Ipaddress]::TryParse($ip, [ref]0)))
     {
-       throw "IP Address *$IPAddress* is not in the correct format. Please correct the ipaddress in the configuration and try again"
+        throw ( @(
+            "IP Address *$IPAddress* is not in the correct format. Please correct the ipaddress "
+            "in the configuration and try again"
+            ) -join "" | Out-String
+        )
     }
 
     try
     {
         Write-Verbose -Message "Checking the IPAddress ..."
         #Get the current IP Address based on the parameters given.
-        $currentIP = Get-NetIPAddress -InterfaceAlias $InterfaceAlias -AddressFamily $AddressFamily -ErrorAction Stop
+        $currentIP = Get-NetIPAddress -InterfaceAlias $InterfaceAlias -AddressFamily `
+            $AddressFamily -ErrorAction Stop
+
+        # Build hash table of current settings
+        $currentSettings = @{}
+        $currentSettings['IPAddress'] = $currentIP.IPAddress
+        $currentSettings['PrefixLength'] = $currentIP.PrefixLength
+        $currentSettings['InterfaceAlias'] = $currentIP.InterfaceAlias
+
+        # Flag to signal whether settings are correct
+        $requiresChanges = $false
 
         #Test if the IP Address passed is equal to the current ip address
-        if(!$currentIP.IPAddress.Contains($IPAddress))
+        if(!$currentSettings['IPAddress'].Equals($IPAddress))
```

Comment:
Oh, *that's* probably why it was `contains` before. For your specific scenario, though, the `Get-NetIPAddress` command is scoped to a single `InterfaceAlias`, which should limit the number of IP addresses returned. Multiple IPs would only be present if multiple IPs were configured on a single IP interface. Possible, of course, so let me do some testing on that one...